lib/fpm/cookery/utils.rb

```diff
@@ -85,6 +85,12 @@ def patch(src, level = 0)
         raise "#{src} does not exist" unless File.exist? src
         safesystem "patch -p#{level} --batch < #{src}"
       end
+
+      def go(*args)
+        args = argument_build(*args)
+        args.each {|a| a == '--mod=vendor' && ENV['GO111MODULE'] = 'on' }
+        safesystem "go #{args.join(' ')}"
```

Comment:
Previously I had this line like:

```
safesystem "go", *args
```
but would end up with the following error:

```
fpm-cookery-0.33.0/lib/fpm/cookery/utils.rb:10:in `system': no implicit conversion of Symbol into String (TypeError)
```

So the join is my crappy solution. Happy for other suggested workarounds.

Reply:
This is because `system` cannot automatically convert symbols to strings. We can do the following to handle this:

```diff
diff --git lib/fpm/cookery/utils.rb lib/fpm/cookery/utils.rb
index b6b2485..cd12619 100644
--- lib/fpm/cookery/utils.rb
+++ lib/fpm/cookery/utils.rb
@@ -6,8 +6,13 @@ module FPM
       protected
       # From fpm. (lib/fpm/util.rb)
       def safesystem(*args)
-        # Make sure to avoid nil elements in args. This might happen on 1.8.
-        success = system(*args.compact.flatten)
+        # Process the arguments:
+        #   - Convert all arguments to strings because "system" cannot handle
+        #     keys (avoids TypeError)
+        #   - Make sure to avoid nil elements in args. This might happen on 1.8.
+        safe_args = args.map(&:to_s).compact.flatten
+        Log.debug("Executing command: #{safe_args.inspect}")
+        success = system(*safe_args)
         if !success
           raise "'system(#{args.inspect})' failed with error code: #{$?.exitstatus}"
         end
@@ -89,7 +94,7 @@ module FPM
       def go(*args)
         args = argument_build(*args)
         args.each {|a| a == '--mod=vendor' && ENV['GO111MODULE'] = 'on' }
-        safesystem "go #{args.join(' ')}"
+        safesystem "go", *args
       end
     end
   end
```

What do you think?

Reply:
@bernd sounds good :ok_hand: I've added your patch and its passing tests now.